File: Python/1572. Matrix Diagonal Sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        result = 0
        for i in range(len(mat)): result += mat[i][i] + mat[i] [::-1][i]
        if len(mat) % 2 == 1: result -= mat[len(mat) // 2][len(mat) // 2]
        return result

File: C/1572. Matrix Diagonal Sum.c
int diagonalSum(int** mat, int matSize, int* matColSize){
    int res = 0;
    int c = matSize;

    for (int i = 0; i < c;i ++) {
        res += mat[i][i] + mat[c - 1 - i][i];
    }

    if (matSize % 2) res -= mat[matSize / 2][matSize / 2];

    return res;
}
